main/ejml-dsparse/src/org/ejml/sparse/csc/CommonOps_DSCC.java

```diff
@@ -1835,5 +1837,111 @@ public static double trace( DMatrixSparseCSC A ) {
 
         return output;
     }
+
+    /**
+     * This applies a given unary function on every value stored in the matrix
+     *
+     * @param input  (Input) input matrix. Not modified
+     * @param func   Unary function accepting a double
+     * @param output (Input/Output) Matrix. Modified.
+     */
+    public static void apply(DMatrixSparseCSC input, DoubleUnaryOperator func, @Nullable DMatrixSparseCSC output) {
```

Comment:
you should return output. If it's null then this function does nothing.

Reply:
ah right. Thanks for spotting this!